lib/src/contexts/variables_source/dns.rs

```diff
@@ -0,0 +1,25 @@
+use std::collections::HashMap;
+
+use anyhow::Result;
+use reqwest::Url;
+use trust_dns_resolver::config::ResolverConfig;
+use trust_dns_resolver::config::ResolverOpts;
+use trust_dns_resolver::Resolver;
+
+pub fn txt_record_values(url: &Url, contexts: &mut HashMap<String, String>) -> Result<()> {
+    let resolver = Resolver::new(ResolverConfig::default(), ResolverOpts::default())?;
+
+    let host = url
+        .host_str()
+        .ok_or_else(|| anyhow::anyhow!("Failed to parse host"))?;
+
+    let records = resolver.txt_lookup(host)?;
+
+    for record in records {
+        if let Some((key, value)) = record.to_string().split_once('=') {
+            contexts.insert(key.to_string(), value.to_string());
```

Comment:
Not for this PR, but to think about later. Variables where `BTreeMap<String, String>` because their first implementation was rather basic.

However, Contexts do accept `<String, Value>` and we could get much smarter about nested objects within TXT/SRV records, or even other sources such as TOML and YAML.

We can expand this

Reply:
True that. I'll see how I can incorporate this in the current PR. Less work in future PRs :P

Reply:
https://github.com/comtrya/comtrya/issues/211